### ISSUE TITLE
[BUG] Code snippet not available in XML files when a match is the key of the node itself and not the value of it

### ISSUE THREAD

**Issue by pranavgaikwad** [konveyor/analyzer-lsp#503]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

v0.3.0

### Priority

Minor

### Current Behavior

No code snippet available for match from _beans.xml_ file in the project.

Example rule: 
```yaml
- category: potential
  customVariables: []
  description: '`beans.xml` descriptor content is ignored'
  effort: 3
  labels:
  - konveyor.io/source=java-ee
  - konveyor.io/target=quarkus
  links: []
  message: "\n `beans.xml` descriptor content is ignored and it could be removed from
    the application. \n Refer to the guide referenced below to check the supported
    CDI feature in Quarkus.\n "
  ruleID: cdi-to-quarkus-00030
  when:
    as: root
    builtin.xml:
      filepaths:
      - beans.xml
      namespaces:
        b: http://xmlns.jcp.org/xml/ns/javaee
      xpath: /b:beans
```

Example app: https://github.com/WASdev/sample.daytrader7

### Expected Behavior

Code snippet around Line no 17 should be present.

### How Reproducible

Always (Default)

### Steps To Reproduce

See example rule and app above, it should return code snippet surrounding line 17

### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```
